lib/ipfs/IpfsStorage.ts

```diff
@@ -44,8 +42,48 @@ export default class IpfsStorage {
     return IpfsStorage.ipfsStorageSingleton;
   }
 
-  private constructor (node: IPFS) {
+  private constructor (node: IPFS, repo: any) {
+    this.repo = repo;
     this.node = node;
+    this.healthy = true;
+    this.healthCheckInternalInSeconds = 60;
+  }
+
+  private static async getNode (repo?: any): Promise<IPFS> {
+    const localRepoName = 'sidetree-ipfs';
+    const options = {
+      repo: repo !== undefined ? repo : localRepoName
+    };
+    const node = await IPFS.create(options);
+    return node;
+  }
+
+  /**
+   * performs health check once every minute
```

Comment:
Nit: Get rid off reference to "every minute":
```suggestion
   * Start periodic health check.
```